public/js/reducers/tabs.js

```diff
@@ -47,4 +60,5 @@ function getTabId(tab) {
   return id;
 }
 
+const update = setupReducer(initialState, { addTabs, selectTab });
```

Comment:
After thinking about it for a couple of days, I'm on the fence between two approaches:

1. the delegator factory
```js
setupReducer(initialState, { addTabs, selectTab });
```

2. an update function with separate handlers
```js
function update(state=InitialState, action)
  switch (action.type) {
    case constants.ADD_TABS:
      return addTab(state, action);
     default:
       return state
  }
```

This second pattern is what [tofino](https://github.com/mozilla/tofino/blob/master/app/ui/browser-modern/reducers/ui.js#L19-L51) does. Benefits:
 * similar benefits with helper functions (documentation, easier to test, ...)
 * more familiar to a larger audience that is used to the the `update/case statement` redux style
 * it's easy to have many action types map to the same helper

As this is a style question, I'd prefer something that is comfortable to a larger audience (option 2) unless we believe enough people would recognize/understand option 1.  I've personally seen a couple projects use option 1, but I don't have a good feeling for how widely adopted it is.